### PR TITLE
Add introduction to programming

### PR DESCRIPTION
**Pull request by core-man**

### CODE REVIEW

File: source/programming/introduction.rst [hunk 1]
```diff
@@ -1,2 +1,28 @@
...
+
+日常的数据处理和简单的科学计算
+可以使用脚本语言快速实现，以便迅速上手和开展研究。如果需要更高的计算效率，可以
+使用编译语言编写代码。因此，我们在完成一个项目的时候，往往是同时使用这两类计算机语言
+来编写代码。有时候还可以用脚本语言快速、便捷地实现一个任务的一部分，然后调用使用编译语言
+书写的需求计算效率的另外一部分的代码来加速计算。
+
+需要注意的是，在本教程中我们将 Shell 也归为脚本语言。严格说来，Linux 和 maxOS 系统的
+Shell 其实本身只是一个空壳，具有最基本的条件判断和循环功能。借助 Linux 系统下命令行工具，
+如 ``awk``\ 、\ ``sort``\ 、\ ``grep``\ ，其可以用来做日常的数据处理、字符串处理等。
```
Comment (core-man) [inline source/programming/introduction.rst:25]:
```suggestion
需要注意的是，在本教程中我们将Linux 和 maxOS 系统的 Shell 也归为脚本语言。严格说来，
Shell 其实本身只是一个空壳，具有最基本的条件判断和循环功能。借助 Linux 和 maxOS 系统下的
命令行工具，如 ``awk``\ 、\ ``sort``\ 、\ ``grep``\ ，其可以用来做日常的数据处理、字符串处理等。
```

File: source/programming/introduction.rst [hunk 2]
```diff
@@ -1,2 +1,28 @@
 前言
 ====
+
+日常科研工作中，我们需要使用计算机语言编写程序和脚本，以自动完成相应的数据处理或科学计算。
+因此，需要了解\ **常见的计算机语言**\  、\ **编译和链接**\ 以及\ **版本控制**\ 等。
```
Comment (seisman) [inline source/programming/introduction.rst:5]: 计算机语言？编程语言？
```suggestion
因此，需要了解\ **常见的编程语言**\  、\ **编译和链接**\ 以及\ **版本控制**\ 等。
```

File: source/programming/introduction.rst [hunk 3]
```diff
@@ -1,2 +1,28 @@
...
+
+计算机语言一般可以分为两大类。第一类是需要编译和链接的\ **编译语言**\ ，如 C 语言、
+Fortran 语言；第二类是不需编译便可直接执行的\ **解释型语言**\ ，也称\ **脚本语言**\ ，如
+Python 语言、Julia 语言。通常来说，使用编译语言编写的代码运行速度更快，
+但是编写代码的时间成本更多，且需要我们了解更多的编程基础；脚本语言易于编写，
+开源的软件库和模块更多，但是运行速度较慢。
+
+我们推荐至少掌握一门编译语言，同时至少掌握一门脚本语言。使用哪种编译语言并不重要，熟悉掌握一门
+编译语言，其他编译语言也就可以很快地学会和上手了。至于脚本语言，我们推荐使用目前至少是在地震学领域
+最流行的 Python 语言，其官方模块、第三方模块和软件包众多，基本可以满足我们日常的科研工作需求。
```
Comment (seisman) [inline source/programming/introduction.rst:15]: We should rethink it. I have the feeling that people now can do researches using Python only, even though they know nothing about C and Fortran. So perhaps we should say "a scripting language is required, but a compiling language is also recommended".

Comment (core-man) [reply source/programming/introduction.rst:15]: okay

File: source/programming/introduction.rst [hunk 4]
```diff
@@ -1,2 +1,36 @@
...
 ====
+
+:本节作者: `姚家园 <https://github.com/core-man>`__\ （作者）、
+          `田冬冬 <https://me.seisman.info/>`__\ （作者）
+:最近更新日期: 2021-01-13
+:预计阅读时间: 2 分钟
+
+----
+
+日常科研工作中，我们需要使用计算机语言编写程序和脚本，以自动完成相应的数据处理或科学计算。
```
Comment (seisman) [inline source/programming/introduction.rst:11]: Please fix all "计算机语言" to "编程语言".

Comment (core-man) [reply source/programming/introduction.rst:11]: Done~

File: source/programming/introduction.rst [hunk 5]
```diff
@@ -1,2 +1,36 @@
...
+          `田冬冬 <https://me.seisman.info/>`__\ （作者）
+:最近更新日期: 2021-01-13
+:预计阅读时间: 2 分钟
+
+----
+
+日常科研工作中，我们需要使用编程语言编写程序和脚本，以自动完成相应的数据处理或科学计算。
+因此，需要了解\ **常见的编程语言**\  、\ **编译和链接**\ 以及\ **版本控制**\ 等。
+
+编程语言一般可以分为两大类。第一类是需要编译和链接的\ **编译语言**\ ，如 C 语言、
```
Comment (seisman) [inline source/programming/introduction.rst:14]:
```suggestion
编程语言一般可以分为两大类。第一类是需要编译和链接的\ **编译型语言**\ ，如 C 语言、
```

File: source/programming/introduction.rst [hunk 6]
```diff
@@ -1,2 +1,36 @@
...
+Fortran 语言；第二类是不需编译便可直接执行的\ **解释型语言**\ ，也称\ **脚本语言**\ ，如
+Python 语言、Julia 语言。通常来说，使用编译语言编写的代码运行速度更快，
```
Comment (seisman) [inline source/programming/introduction.rst:16]:
```suggestion
Python 语言、Julia 语言。通常来说，使用编译型语言编写的代码运行速度更快，
```

File: source/programming/introduction.rst [hunk 7]
```diff
@@ -1,2 +1,36 @@
...
+但是编写代码的时间成本更多，且需要我们了解更多的编程基础；脚本语言易于编写，
```
Comment (seisman) [inline source/programming/introduction.rst:17]:
```suggestion
但是编写代码的时间成本更多，且需要掌握更多的编程技能；脚本语言易于编写，
```

File: source/programming/introduction.rst [hunk 8]
```diff
@@ -1,2 +1,36 @@
...
+开源的软件库和模块更多，但是运行速度较慢。
+
+**地震学科研工作者需至少掌握一门脚本语言，同时我们推荐学习一门编译语言**\ 。对于脚本语言，
```
Comment (seisman) [inline source/programming/introduction.rst:20]:
```suggestion
**地震学科研工作者需至少掌握一门脚本语言，同时我们推荐学习一门编译型语言**\ 。对于脚本语言，
```

File: source/programming/introduction.rst [hunk 9]
```diff
@@ -1,2 +1,36 @@
...
+我们推荐使用目前至少是在地震学领域最流行的 Python 语言，其官方模块、第三方模块和软件包众多，
```
Comment (seisman) [inline source/programming/introduction.rst:21]:
```suggestion
我们推荐使用目前在地震学领域最流行的 Python 语言，其官方模块、第三方模块和软件包众多，
```

File: source/programming/introduction.rst [hunk 10]
```diff
@@ -1,2 +1,36 @@
...
+基本可以满足我们日常的科研工作需求。而使用哪种编译语言并不重要，熟悉掌握一门编译语言，
+其他编译语言也就可以很快地学会和上手了。
```
Comment (seisman) [inline source/programming/introduction.rst:23]:
```suggestion
基本可以满足我们日常的科研工作需求。而使用哪种编译型语言并不重要，熟悉掌握一门编译型语言，
其他编译型语言也就可以很快地学会和上手了。
```

File: source/programming/introduction.rst [hunk 11]
```diff
@@ -1,2 +1,36 @@
...
+
+日常的数据处理和简单的科学计算
+可以使用脚本语言快速实现，以便迅速上手和开展研究。如果需要更高的计算效率，可以
+使用编译语言编写代码。因此，我们在完成一个项目的时候，往往是同时使用这两类编程语言
```
Comment (seisman) [inline source/programming/introduction.rst:27]:
```suggestion
使用编译型语言编写代码。因此，我们在完成一个项目的时候，往往是同时使用这两类编程语言
```

File: source/programming/introduction.rst [hunk 12]
```diff
@@ -1,2 +1,36 @@
...
+来编写代码。有时候还可以用脚本语言快速、便捷地实现一个任务的一部分，然后调用使用编译语言
```
Comment (seisman) [inline source/programming/introduction.rst:28]:
```suggestion
来编写代码。有时候还可以用脚本语言快速、便捷地实现一个任务的一部分，然后调用使用编译型语言
```

**Review by seisman — APPROVED** [review]

**Pull request merged**